packages/components/src/Persona.purs

```diff
@@ -434,3 +437,116 @@ instance writeForeignDeliveryReclamationStatus :: WriteForeign DeliveryReclamati
   writeImpl = genericEncodeEnum { constructorTagTransform: \x -> x }
 instance showDeliveryReclamationStatus :: Show DeliveryReclamationStatus where
   show = genericShow
+
+data PaymentState
+  = PaymentOpen
+  | PartiallyPaid
+  | Paid
+  | Reminded
+  | Foreclosure
+  | Reimbursed
+  | CreditLoss
+
+instance readPaymentState :: Read PaymentState where
+  read s =
+    case s of
+      "PaymentOpen"   -> pure PaymentOpen
+      "PartiallyPaid" -> pure PartiallyPaid
+      "Paid"          -> pure Paid
+      "Reminded"      -> pure Reminded
+      "Foreclosure"   -> pure Foreclosure
+      "Reimbursed"    -> pure Reimbursed
+      "CreditLoss"    -> pure CreditLoss
+      _               -> Nothing
+
+data PaymentType
+  = NormalState
+  | DirectDebit
+  | Reminder1
+  | Reminder2
+  | Nonpayment
+  | Reimbursement
+
+instance readPaymentType :: Read PaymentType where
+  read t =
+    case t of
+      "NormalState"   -> pure NormalState
+      "DirectDebit"   -> pure DirectDebit
+      "Reminder1"     -> pure Reminder1
+      "Reminder2"     -> pure Reminder2
+      "Nonpayment"    -> pure Nonpayment
+      "Reimbursement" -> pure Reimbursement
+      _               -> Nothing
+
+type SubscriptionPayments =
+  { name :: String
+  , startDate :: Date
+  , lastDate :: Date
+  , payments :: Array Payment
+  }
+
+type Payment =
+  { date :: Date
+  , dueDate :: Date
+  , expenses :: Number
+  , interest :: Number
+  , vat :: Number
+  , amount :: Number
+  , openAmount :: Number
+  , type :: PaymentType
+  , state :: PaymentState
+  , discount :: Number
+  }
+
+type SubscriptionPayments' =
```

Comment:
Very nice to separate the js/json type from the type we're using in code. We should do this for other types as well where it makes sense. However, could we call this `ApiSubscriptionPayments` or `JSSubscriptionPayments` or `SubscriptionPaymentsJS` or something similar? We do have e.g. `JSProps` in some of our react components, so `JS` prefix or postfix would be in line with those 🤔

Reply:
I changed it to have `Api` prefix. It's not used outside of this file and I wouldn't expect it to be either.